### DISCUSSION TITLE
Wconf: support src filter from Scala 2

### DISCUSSION THREAD

**Discussion by kubukoz** [scala/scala3#17635] [Feature Requests]
In https://github.com/lampepfl/dotty/pull/12857, the `-Wconf` options were added. However, the `src` filter, which some people are using to [ignore warnings from generated code](https://stackoverflow.com/a/66354074), has seemingly been left out.

I'm getting this error when trying to use it:

```
[error] [core3] Failed to parse `-Wconf` configuration: src=src_managed/.*:silent
[error] [core3] unknown filter: src
```

In my eyes it's more of a bug report than a feature request (since the feature exists in the 2.x versions of the language, and its lack of existence is a potential migration blocker), but I'm reporting here as a first step.

**Comment by lrytz** [comment]
I'm happy to add it if noone objects.

**Comment by pribor** [comment]
Any progress on this request?

**Comment by tOverney** [comment]
@lrytz would be great to have it :)

**Comment by dzanot** [comment]
Might I also suggest the `site` and `origin` filters as well.

**Reply by ngbinh** [reply]
Seconded that src filter is quite important for us.

**Comment by austek** [comment]
I can work on this, however this will be my first contribution to this codebase and might need some guidance

**Comment by povder** [comment]
I went ahead and created a PR https://github.com/lampepfl/dotty/pull/18783 after creating https://github.com/lampepfl/dotty/issues/18782 issue

**Comment by Gedochao** [comment]
Closed in https://github.com/scala/scala3/pull/18783, will be included in 3.5.0-RC1.
Closing this discussion.

**Reply by kubukoz** [reply]
Any chance we can have this in LTS as well?

**Reply by povder** [reply]
> Any chance we can have this in LTS as well?

https://github.com/scala/scala3/pull/18783#issuecomment-2077003536
